dev/devicelab/bin/tasks/gradle_jetifier_test.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/apk_utils.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:path/path.dart' as path;
+
+final String gradlew = Platform.isWindows ? 'gradlew.bat' : 'gradlew';
+final String gradlewExecutable = Platform.isWindows ? gradlew : './$gradlew';
+
+/// Tests that Jetifier can translate plugins that use support libraries.
+Future<void> main() async {
+  await task(() async {
+
+    section('Find Java');
+
+    final String javaHome = await findJavaHome();
+    if (javaHome == null)
+      return TaskResult.failure('Could not find Java');
+    print('\nUsing JAVA_HOME=$javaHome');
+
+    section('Create Flutter AndroidX app project');
+
+    final Directory tempDir = Directory.systemTemp.createTempSync('flutter_module_test.');
+    final Directory projectDir = Directory(path.join(tempDir.path, 'hello'));
+    try {
+      await inDirectory(tempDir, () async {
+        await flutter(
+          'create',
+          options: <String>[
+            '--org', 'io.flutter.devicelab',
+            '--androidx',
+            'hello',
+          ],
+        );
+      });
+
+      section('Add plugin that uses support libraries');
+
+      final File pubspec = File(path.join(projectDir.path, 'pubspec.yaml'));
+      String content = await pubspec.readAsString();
+      content = content.replaceFirst(
+        '\ndependencies:\n',
+        '\ndependencies:\n  firebase_auth: 0.7.0\n',
+      );
+      await pubspec.writeAsString(content, flush: true);
+      await inDirectory(projectDir, () async {
+        await flutter(
+          'packages',
+          options: <String>['get'],
+        );
+      });
+
+      section('Build release APK');
+
+      await inDirectory(projectDir, () async {
+        await flutter(
+          'build',
+          options: <String>[
+            'apk',
+            '--target-platform', 'android-arm',
+            '--verbose',
+          ],
+          environment: <String, String>{'ENABLE_FLUTTER_BUILD_PLUGINS_AS_AAR': 'true'},
+        );
+      });
+
+      final File releaseApk = File(path.join(
+        projectDir.path,
+        'build',
+        'app',
+        'outputs',
+        'apk',
+        'release',
+        'app-release.apk',
+      ));
+
+      if (!exists(releaseApk)) {
+        return TaskResult.failure('Failed to build release APK.');
+      }
+
+      checkApkContainsClasses(releaseApk, <String>[
+        // The plugin class defined by `firebase_auth`.
+        'io.flutter.plugins.firebaseauth.FirebaseAuthPlugin',
+        // Used by `firebase_auth`.
+        'com.google.firebase.FirebaseApp',
```

Comment:
it's too late to check for `android.support.*` here, since all previous references are now "androidX" equivalent. I did add a check for some androidx dependencies.